### ISSUE TITLE
pgfplots fillrange does not work 

### ISSUE THREAD

**Issue by arturgower** [JuliaPlots/Plots.jl#1216]
The option fillrange does not work with pgfplots()
```
using Plots
pgfplots()
y= 1 + rand(30)
plot(y, fillrange = 0)
```
The results is:
![julia7cqtfv](https://user-images.githubusercontent.com/24454863/32234434-ae72efa2-be54-11e7-8ffe-ebb690ab7135.jpg)

whereas the graph should have been filled in from 0 to the curve y, as in 
![figure_1](https://user-images.githubusercontent.com/24454863/32234491-dbf9293c-be54-11e7-86ed-e920b545863b.png)
 
Anyone know how to fix this?

**Comment by apalugniok** [comment]
I can recreate this with PGFPlots.jl, seems like PGFPlots fills between the y data and the line joining the first and last point of the data. Not sure how to hack this to work as it should 😟

**Comment by mkborregaard** [comment]
It'd be nice to get this fixed on pgfplots. It doesn't look too complicated.

**Comment by sivapvarma** [comment]
@apalugniok looks like what you described is correct. it fills between y data and line joining first and last point of data.

```julia
using Plots
pgfplots()

x = 0:0.01:π/2
plot(x, sin.(x), line=2, fill=(0, 0.5, :blue))
plot!(x, x, line=2, fill=(0, 0.5, :green))
```

![out](https://user-images.githubusercontent.com/1314675/32424409-226e6246-c261-11e7-8323-66180341df34.png)

**Comment by sivapvarma** [comment]
this is my current understanding of this issue.  In PGFPlots backend, fillrange option does not have any effect. it simply goes to the style.  See example plts below.

```julia
using Plots
pgfplots()

plot(0:10, (0:10).^2, fill=(0,0.5,:blue))
```

![plots](https://user-images.githubusercontent.com/1314675/32478049-3b5b1aca-c337-11e7-926b-c3e7f2b07a61.png)

```julia
import PGFPlots
p = PGFPlots.Plots.Linear(0:10, (0:10).^2, 
    style="red, fill=blue, fill opacity=0.5", mark="none")
```

![pgfplots](https://user-images.githubusercontent.com/1314675/32478172-0ac7cb32-c338-11e7-922b-ff1670905cf7.png)

To fill between fillrange and y correctly as intended an additional `\addplot` has to be sent  to tikzpicture like [here](https://tex.stackexchange.com/a/163655/9609) and [here](https://tex.stackexchange.com/a/163654/9609) which will need that additional `\addplot` and `fillbetween` pdfplotslibrary.  Does PGFPlots.jl support  adding  `\addplot` for the fill ? If yes, we need a way to name the paths.

Is there a better way ?

**Comment by apalugniok** [comment]
You can use `PGFPlots.Command()` to add any type of `PFPlots TeX` code to the `Axis`. From the examples I've seen you could use `\addpath` to specify the other part of the `fill`. For the path names I currently don't have a better idea than name then `series` and `fillpath`and make sure that `fillbetween` is used before the next series is added, but that seems really unsafe so I'm not a fan of that naming scheme.

**Comment by mschauer** [comment]
A workaround is 
```
struct CFBand
     tt
     yy
     vv
end

@recipe function plot(X::CFBand; quantile = 0.95)
    q = Distributions.quantile(Normal(), 1 - (1 - quantile)/2)
    if Plots.backend() != Plots.PGFPlotsBackend()
        ribbon --> q*sqrt.(X.vv)
        X.tt, X.yy
    else
        ribbon --> (NaN,NaN)
        linewidth --> 0
        vcat(X.tt, reverse(X.tt)), vcat(X.yy + q*sqrt.(X.vv), reverse(X.yy - q*sqrt.(X.vv)))
    end
end
```

**Comment by mkborregaard** [comment]
It should be fixable in the backend code though.

**Comment by daschw** [comment]
fixed by #1467
